xdsl/ir.py

```diff
@@ -139,11 +139,11 @@ def __repr__(self) -> str:
                f"op_name={repr(self.op.name)}, " + \
                f"result_index={repr(self.result_index)}, name={repr(self.name)})"
 
-    def __eq__(self, other: OpResult) -> bool:
+    def __eq__(self, other: object) -> bool:
```

Comment:
This is a type hint enforced by the override of `__eq__`, right? I am not quite happy with this new form. Maybe add an instance check, s.t. it does the same thing if the input is an OpResult, o/w it should crash IMO. Comparing equality on something weird with an OpResult feels wrong each time.

Reply:
Yeah, it seems that what was before was broken already.
So the choice is either:
* Do nothing, in that case we can compare things that are of different types
* Trigger an exception.

Actually, I think that only the first option makes sense.
What if we want to compare an `OpResult` to an `SSAValue`? In that case, it makes sense to keep object as input (or at least `SSAValue`)